parachain-template/runtime/src/xcm_config.rs

```diff
@@ -87,12 +87,19 @@ match_types! {
 	};
 }
 
-pub type Barrier = (
-	TakeWeightCredit,
-	AllowTopLevelPaidExecutionFrom<Everything>,
-	AllowUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
-	// ^^^ Parent and its exec plurality get free execution
-);
+#[path = "../../../polkadot-parachains/parachains-common/src/xcm_config.rs"]
```

Comment:
What is this syntax here?

Reply:
Why is the template modified at all?

Reply:
> Why is the template modified at all?

Because of this: https://github.com/paritytech/cumulus/pull/1169#issuecomment-1097908080

Reply:
Why can a parachain not act as a reserve for another parachain?

Imo this makes only sense when you assume that reserves are done on the relay chain.

Reply:
Hmm, so the problem here is that the relay chain recognizes absolutely no chain to be a reserve, and so any parachain _should not_ be sending `ReserveAssetDeposited` _towards the relay chain_, because that's what we are specifically banning here.

I'm still weirded out by the syntax used here however. Why can't we just import the Rust module properly via Cargo.toml?

Reply:
> Hmm, so the problem here is that the relay chain recognizes absolutely no chain to be a reserve, and so any parachain _should not_ be sending `ReserveAssetDeposited` _towards the relay chain_ as a result of executing `DepositReserveAsset` or `TransferReserveAsset` locally, because that's what we are specifically banning here.

Ahh yeah sorry. It is matched for the sending stuff upwards.

> I'm still weirded out by the syntax used here however. Why can't we just import the Rust module properly via Cargo.toml?

Yeah, this will not be accepted.

Reply:
Dan wasn't so keen on the idea of the template depending on the parachain common code. If you don't like the above, I can brake it out into a separate very-common crate or I can duplicate the code or I could just not include it in the template. Happy to do whatever we can agree on.

Reply:
For now I have duplicated the code. Some of that code will be removed once `DenyThenTry` gets pushed down into polkodot's xcm module.